nwb_conversion_tools/json_schema_utils.py

```diff
@@ -67,10 +68,24 @@ def get_schema_from_method_signature(class_method, exclude=None):
     for param in inspect.signature(class_method).parameters.values():
         if param.name not in exclude + ['self']:
             if param.annotation:
-                param_type = annotation_json_type_map[str(param.annotation).split("'")[1]]
+                anno = str(param.annotation)
+                if "typing.Union" in anno:
+                    types = re.search("typing.Union\[(.*)\]", anno).group(1).split(",")
+                    intersect_valid_keys = list(set(annotation_json_type_map.keys()).intersection(types))
```

Comment:
```suggestion
                anno = str(param.annotation)
                if "typing.Union" in anno:
                    types = re.search("typing.Union\[(.*)\]", anno).group(1).split(",")
                    intersect_valid_keys = list(set(annotation_json_type_map).intersection(types))
```

Reply:
Right you are; I keep forgetting `set `only casts the `keys` of a `dict`.